vectorstores/alloydb/engine.go

```diff
@@ -0,0 +1,148 @@
+package alloydb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+
+	"cloud.google.com/go/alloydbconn"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/idtoken"
+	"google.golang.org/api/oauth2/v2"
+	"google.golang.org/api/option"
+)
+
+type EmailRetreiver func(context.Context) (string, error)
+
+type PostgresEngine struct {
+	conn *pgxpool.Pool
+}
+
+// NewPostgresEngine creates a new PostgresEngine.
+func NewPostgresEngine(ctx context.Context, opts ...Option) (*PostgresEngine, error) {
+	pgEngine := new(PostgresEngine)
+	cfg := &engineConfig{
+		port:           defaultPort,
+		emailRetreiver: getServiceAccountEmail,
+	}
+	for _, opt := range opts {
+		opt(cfg)
+	}
```

Comment:
Does there need to be some kind of validation to make sure the right stuff is set here?